tools/src/main/java/org/apache/kafka/trogdor/workload/SustainedConnectionWorker.java

```diff
@@ -0,0 +1,532 @@
+        }
+    }
+
+    public class MaintainLoop implements Runnable {
+        @Override
+        public void run() {
+            try {
+                while (!doneFuture.isDone()) {
+                    Optional<SustainedConnection> currentConnection = SustainedConnectionWorker.this.findConnectionToMaintain();
+                    if (currentConnection.isPresent()) {
+                        currentConnection.get().refresh();
+                    } else {
+                        SustainedConnectionWorker.time.sleep(SustainedConnectionWorker.BACKOFF_PERIOD_MS);
+                    }
+                }
+            } catch (Exception e) {
+                SustainedConnectionWorker.this.totalAbortedThreads.incrementAndGet();
+                SustainedConnectionWorker.log.error("Aborted thread while maintaining sustained connections", e);
+            }
+        }
+    }
+
+    private Optional<SustainedConnection> findConnectionToMaintain() {
+        synchronized (SustainedConnectionWorker.this) {
+            for (SustainedConnection connection : this.connections) {
+                if (connection.needsRefresh()) {
+                    connection.claim();
+                    return Optional.of(connection);
+                }
+            }
+        }
+        return Optional.empty();
+    }
+
+    private class StatusUpdater implements Runnable {
+        @Override
+        public void run() {
+            try {
+                JsonNode node = JsonUtil.JSON_SERDE.valueToTree(
+                    new StatusData(
+                            SustainedConnectionWorker.this.totalProducerConnections.get(),
+                            SustainedConnectionWorker.this.totalProducerFailedConnections.get(),
+                            SustainedConnectionWorker.this.totalConsumerConnections.get(),
+                            SustainedConnectionWorker.this.totalConsumerFailedConnections.get(),
+                            SustainedConnectionWorker.this.totalMetadataConnections.get(),
+                            SustainedConnectionWorker.this.totalMetadataFailedConnections.get(),
+                            SustainedConnectionWorker.this.totalAbortedThreads.get(),
+                            SustainedConnectionWorker.time.milliseconds()));
+                status.update(node);
+            } catch (Exception e) {
+                SustainedConnectionWorker.log.error("Aborted test while running StatusUpdater", e);
+                WorkerUtils.abort(log, "StatusUpdater", e, doneFuture);
+            }
+        }
+    }
+
+    public static class StatusData {
+        private final long totalProducerConnections;
+        private final long totalProducerFailedConnections;
+        private final long totalConsumerConnections;
+        private final long totalConsumerFailedConnections;
+        private final long totalMetadataConnections;
+        private final long totalMetadataFailedConnections;
+        private final long totalAborts;
```

Comment:
This needs to be `totalAbortedThreads`

Reply:
Oops, wonder how it compiled for me like this.  Done.